src/field_specials.c

```diff
@@ -0,0 +1,2503 @@
+            {
+                if (get_unknown_box_id() != i)
+                    FlagClear(FLAG_SYS_CHANGED_BOX_TO_STORE_MON);
+                VarSet(VAR_0x4037, i);
+                return sub_80CC7B4();
+            }
+        }
+        i++;
+        if (i == TOTAL_BOXES_COUNT)
+            i = 0;
+    } while (i != StorageGetCurrentBox());
+    return FALSE;
+}
+
+const u16 sPokeCenter1FMaps[] = {
+    MAP_VIRIDIAN_CITY_POKEMON_CENTER_1F,
+    MAP_PEWTER_CITY_POKEMON_CENTER_1F,
+    MAP_CERULEAN_CITY_POKEMON_CENTER_1F,
+    MAP_LAVENDER_TOWN_POKEMON_CENTER_1F,
+    MAP_VERMILION_CITY_POKEMON_CENTER_1F,
+    MAP_CELADON_CITY_POKEMON_CENTER_1F,
+    MAP_FUCHSIA_CITY_POKEMON_CENTER_1F,
+    MAP_CINNABAR_ISLAND_POKEMON_CENTER_1F,
+    MAP_INDIGO_PLATEAU_POKEMON_CENTER_1F,
+    MAP_SAFFRON_CITY_POKEMON_CENTER_1F,
+    MAP_ROUTE4_POKEMON_CENTER_1F,
+    MAP_ROUTE10_POKEMON_CENTER_1F,
+    MAP_ONE_ISLAND_POKEMON_CENTER_1F,
+    MAP_TWO_ISLAND_POKEMON_CENTER_1F,
+    MAP_THREE_ISLAND_POKEMON_CENTER_1F,
+    MAP_FOUR_ISLAND_POKEMON_CENTER_1F,
+    MAP_FIVE_ISLAND_POKEMON_CENTER_1F,
+    MAP_SEVEN_ISLAND_POKEMON_CENTER_1F,
+    MAP_SIX_ISLAND_POKEMON_CENTER_1F,
+    MAP_UNKNOWN_MAP_00_04,
+    MAP_UNDEFINED
+};
+
+bool8 sub_80CC87C(void)
+{
+    s32 i;
+    u16 mapno = (gUnknown_2031DB4.mapGroup << 8) + gUnknown_2031DB4.mapNum;
+    for (i = 0; sPokeCenter1FMaps[i] != MAP_UNDEFINED; i++)
+    {
+        if (sPokeCenter1FMaps[i] == mapno)
+            return TRUE;
+    }
+    return FALSE;
+}
+
+bool8 Special_ItemIsTM_GetMoveName(void)
+{
+    // 8004 = item ID
+    if (gSpecialVar_0x8004 >= ITEM_TM01 && gSpecialVar_0x8004 <= ITEM_HM08)
+    {
+        StringCopy(gStringVar1, gMoveNames[ItemIdToBattleMoveId(gSpecialVar_0x8004)]);
+        return TRUE;
+    }
+    else
+        return FALSE;
+}
+
+void RunMassageCooldownStepCounter(void)
+{
+    u16 count = VarGet(VAR_MASSAGE_COOLDOWN_STEP_COUNTER);
+    if (count < 500)
+        VarSet(VAR_MASSAGE_COOLDOWN_STEP_COUNTER, count + 1);
+}
+
+void Special_DaisyMassageServices(void)
+{
+    AdjustFriendship(&gPlayerParty[gSpecialVar_0x8004], 6);
+    VarSet(VAR_MASSAGE_COOLDOWN_STEP_COUNTER, 0);
+}
+
+static const u16 sEliteFourLightingPalettes[][16] = {
+    INCBIN_U16("data/field_specials/unk_83F5F50.gbapal"),
+    INCBIN_U16("data/field_specials/unk_83F5F70.gbapal"),
+    INCBIN_U16("data/field_specials/unk_83F5F90.gbapal"),
+    INCBIN_U16("data/field_specials/unk_83F5FB0.gbapal"),
+    INCBIN_U16("data/field_specials/unk_83F5FD0.gbapal"),
+    INCBIN_U16("data/field_specials/unk_83F5FF0.gbapal"),
+    INCBIN_U16("data/field_specials/unk_83F6010.gbapal"),
+    INCBIN_U16("data/field_specials/unk_83F6030.gbapal"),
+    INCBIN_U16("data/field_specials/unk_83F6050.gbapal"),
+    INCBIN_U16("data/field_specials/unk_83F6070.gbapal"),
+    INCBIN_U16("data/field_specials/unk_83F6090.gbapal"),
+    INCBIN_U16("data/field_specials/unk_83F60B0.gbap
```

Comment:
Is it really better to explicitly write != FALSE here?

Reply:
It appears to be the style of this particular programmer, based on his use of `!= TRUE` on the next line